api/envoy/extensions/http/original_ip_detection/xff/v3/xff.proto

```diff
@@ -22,5 +26,37 @@ message XffConfig {
   // determining the origin client's IP address. The default is zero if this option
   // is not specified. See the documentation for
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
+  //
+  // If ``xff_trusted_cidrs`` is configured, this field is not used.
   uint32 xff_num_trusted_hops = 1;
+
+  // The `CIDR <https://tools.ietf.org/html/rfc4632>`_ ranges to trust when
+  // evaluating the remote IP address to determine the origin client's IP address.
+  // This is used instead of
+  // :ref:`use_remote_address <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.use_remote_address>`.
+  // When the remote IP address matches a trusted CIDR and the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` header was sent, the last
+  // IP address from ``x-forwarded-for`` is used as the original client address.
+  // If no ``x-forwarded-for`` header was sent, the remote IP is considered the real IP.
+  //
+  // If ``recurse`` is set the :ref:`config_http_conn_man_headers_x-forwarded-for`
+  // header is also evaluated against the trusted CIDR list, and the last non-trusted
+  // address is used as the original client address.
+  //
+  // This is typically used when requests are proxied by a
+  // `CDN <https://en.wikipedia.org/wiki/Content_delivery_network>`_.
+  //
+  // If this field is configured, ``xff_num_trusted_hops`` is not used.
+  XffTrustedCidrs xff_trusted_cidrs = 2;
```

Comment:
may be worth a (presumably fairly obvious) comment that these checks are n*m where n is the number of trusted ranges and m is the number of xff hops in the incoming header.

Reply:
Good call, added in eb2f7ff10f.